src/Elm/Writer.elm

```diff
@@ -490,7 +490,16 @@ writeExpression (Node range inner) =
             string ("\"" ++ s ++ "\"")
 
         CharLiteral c ->
-            string ("'" ++ String.fromList [ c ] ++ "'")
+            let
+                escape : String
+                escape =
+                    if c == '\t' || c == '\'' || c == '\\' then
+                        "\\"
+
+                    else
+                        ""
+            in
+            string ("'" ++ escape ++ String.fromList [ c ] ++ "'")
```

Comment:
Couldn't we use String.fromChar instead?

Reply:
Good point. I checked where String.fromList gets used and noticed that CharPattern isn't escaped either so I fixed that now.